civis/io/_tables.py

```diff
@@ -629,11 +643,122 @@ def csv_to_civis(filename, database, table, api_key=None, client=None,
     if archive:
         warnings.warn("`archive` is deprecated and will be removed in v2.0.0. "
                       "Use `hidden` instead.", FutureWarning)
+
+    name = path.basename(filename)
     with open(filename, "rb") as data:
-        fut = _import_bytes(data, database, table, client, max_errors,
-                            existing_table_rows, distkey, sortkey1, sortkey2,
-                            delimiter, headers, credential_id,
-                            polling_interval, archive, hidden=hidden)
+        file_id = file_to_civis(data, name, api_key=api_key, client=client)
+        log.info('Uploaded file %s to Civis file %s', filename, file_id)
+        fut = civis_file_to_table(file_id, database, table, api_key=api_key,
+                                  client=client, max_errors=max_errors,
+                                  existing_table_rows=existing_table_rows,
+                                  distkey=distkey,
+                                  sortkey1=sortkey1, sortkey2=sortkey2,
+                                  delimiter=delimiter, headers=headers,
+                                  credential_id=credential_id,
+                                  polling_interval=polling_interval,
+                                  hidden=hidden)
+    return fut
+
+
+@deprecate_param('v2.0.0', 'api_key')
+def civis_file_to_table(file_id, database, table, api_key=None, client=None,
```

Comment:
Do we need to include the deprecated `api_key` here?

Reply:
removed!